third_party/terraform/resources/resource_compute_network_peering.go.erb

```diff
@@ -40,19 +42,40 @@ func resourceComputeNetworkPeering() *schema.Resource {
 				DiffSuppressFunc: compareSelfLinkRelativePaths,
 			},
 			"auto_create_routes": {
-				Type:     schema.TypeBool,
-				ForceNew: true,
-				Optional: true,
-				Default:  true,
+				Type:             schema.TypeBool,
+				Optional:         true,
+				Deprecated:       "auto_create_routes is deprecated use exchange_subnet_routes instead",
+				ForceNew:         true,
+				Default:          true,
 			},
 			"state": {
-				Type:     schema.TypeString,
-				Computed: true,
+				Type:             schema.TypeString,
+				Computed:         true,
 			},
 			"state_details": {
-				Type:     schema.TypeString,
-				Computed: true,
+				Type:             schema.TypeString,
+				Computed:         true,
+			},
+			"exchange_subnet_routes": {
```

Comment:
We did some research into this and it turns out the API only accepts `true` as a value for `exchange_subnet_routes` or `auto_create_routes` (of which only one can be set in a valid request). Also, you can't set `auto_create_routes` if you use the `networkPeering` object.

I think I'd prefer that we always use the `networkPeering` object (in both beta and GA), deprecate `auto_create_routes`, do not add `exchange_subnet_routes`, and always set `exchangeSubnetRoutes` to `true` in the `networkPeering` object in the request.

Reply:
That seems sound, as I can't imagine a situation where you do not need the echangeSubnetRoutes Parameter. I adjusted everything according to your proposal. No exchange_subnet_routes parameter, deprecation warning for auto_create_routes and default true for exchangeSubnetRoutes in the networkPeering object. Usage of the networkPeering object in ga and beta provider. Tests are running smoothly, but I encountered a strange error:

`==> Checking source code against gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google -v -run=TestAccComputeNetworkPeering_basic -timeout 240m -ldflags="-X=github.com/terraform-providers/terraform-provider-google/version.ProviderVersion=acc"
# github.com/terraform-providers/terraform-provider-google/google [github.com/terraform-providers/terraform-provider-google/google.test]
google/resource_dataproc_cluster_iam_test.go:65:3: undefined: serviceAccountCanonicalEmail
google/resource_dataproc_cluster_iam_test.go:78:80: undefined: serviceAccountCanonicalEmail
google/resource_dataproc_job_iam_test.go:63:3: undefined: serviceAccountCanonicalEmail
FAIL	github.com/terraform-providers/terraform-provider-google/google [build failed]`

I had to remove the two test-files to get the network peering tests running. These two files are 5 days old and originate from this commit: https://github.com/GoogleCloudPlatform/magic-modules/pull/1766

It does not affect this PR, just wanted to let you know that there seems to be an issue, maybe it will occur in your testing pipelines too, or it is just me and my setup...